### PR TITLE
feat: add scrollTo functionality to shop, home, and club pages

### PR DESCRIPTION
**Pull request by guptaanshik1**
## Related Issue 

Closes: #640 

#### Describe the changes you've made ✍
When we scroll to the some point on a page and then navigate away to another page then instead of opening the new page from top the page opens from the point where the user was on the previous page.

### CODE REVIEW

File: src/pages/shops/ShopLanding.jsx [hunk 1]
```diff
@@ -21,8 +26,8 @@ const ShopLanding = () => {
             <div className="shop_textdiv">
               <p className="shop_header1">Shop for charity !</p>
               <p className="shop_header2">
-                Welcome to our shop. Anything you buy from here helps out the
-                charities. We don't take anything from the profit.
+                {`Welcome to our shop. Anything you buy from here helps out the
```
Comment (IAmTamal) [inline src/pages/shops/ShopLanding.jsx:29]: Why did you add the `{` ?

**Review by IAmTamal — CHANGES_REQUESTED** [review]
The rest of the additions looks good to me !

**Comment by IAmTamal** [comment]
Also, you do not need to mention any Issue Link when making PRs, just make sure that instead of `Closes: #[issue number that you are working on]` here you replace `[issue number that you are working on]` with your issue's number.

Example #11  --> Like this.
Do make sure to edit.

File: src/pages/display/ClubsPage.jsx [hunk 2]
```diff
@@ -40,8 +43,8 @@ const ClubsPage = () => {
               <Loading />
             ) : (
               <>
-                {clubData.map((club) => {
-                  return <SingleClubEvent key={club._id} club={club} />;
+                {clubData?.map((club) => {
```
[the remark this answers is not in the document]
Comment (guptaanshik1) [reply src/pages/display/ClubsPage.jsx:46]: Thanks ❤️

File: src/pages/shops/ShopLanding.jsx (hunk 1 above)
Comment (guptaanshik1) [reply src/pages/shops/ShopLanding.jsx:29]: yes I was getting linting error on ' so I wrapped it inside curly braces

**Comment by guptaanshik1** [comment]
Hi @IAmTamal I have made that change you have asked for. You can continue with the further process 😊.

File: src/pages/shops/ShopLanding.jsx (hunk 1 above)
Comment (IAmTamal) [reply src/pages/shops/ShopLanding.jsx:29]: Cool, just replace don't with `don&apos;t`

We use the `&apos;` for '

Comment (guptaanshik1) [reply src/pages/shops/ShopLanding.jsx]: I have made the change please check

**Review by IAmTamal — APPROVED** [review]
**Amazing work,**
This looks good to me, will be merged soon 🎉.
